lib/ansible/modules/windows/win_product_facts.ps1

```diff
@@ -57,9 +57,59 @@ if (-not $product_key) {
     }
 }
 
+try {
+    $license_info = Get-CimInstance SoftwareLicensingProduct | Where-Object PartialProductKey
+
+    switch ($license_info.LicenseStatus){
```

Comment:
You don't have to change it but this can easily be simplified with

```
$winlicense_status = switch ($license_info.LicenseStatus) {
    0 { "Unlicensed" }
    1 { "Licensed" }
    2 { "OOBGrace" }
    ...
    default { "NA" }
}
```

The benefits of this approach is that you only need to set the var name once meaning less complication.